### PR TITLE
amrex: fix missing hdf5 dependency

### PR DESCRIPTION
**Pull request by mic84**

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
LGTM.

Approval pending passing required checks.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Please fix the `flake8` errors:

- var/spack/repos/builtin/packages/amrex/package.py:86: [E231] missing whitespace after ','
15
- var/spack/repos/builtin/packages/amrex/package.py:86: [E231] missing whitespace after ','

File: var/spack/repos/builtin/packages/amrex/package.py [hunk 1]
```diff
@@ -83,6 +83,7 @@ class Amrex(CMakePackage):
     depends_on('cmake@3.14:', type='build', when='@19.04:')
     # cmake @3.17: is necessary to handle cuda @11: correctly
     depends_on('cmake@3.17:', type='build', when='^cuda @11:')
+    depends_on('hdf5@1.10.4:',type='build',when='+hdf5')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/amrex/package.py]:
```suggestion
    depends_on('hdf5@1.10.4:', type='build', when='+hdf5')
```

**Comment by eschnett** [comment]
I think the `type=build` is wrong; it shouldn't be there.

Which HDF5 bindings do you require (just C, or also C++)? Do you require parallel I/O?

**Comment by mic84** [comment]
@eschnett it should be good now

**Comment by eschnett** [comment]
I confirm that this PR works for me.

**Comment by eschnett** [comment]
Ping! This patch is almost good to go, just needs a rebase or merge.

**Comment by mic84** [comment]
@tldahlgren can you review again, please? Thanks!

**Review by tldahlgren — APPROVED** [review]
LGTM.

**Comment by tldahlgren** [comment]
Thank you for your contribution!

**Pull request merged**